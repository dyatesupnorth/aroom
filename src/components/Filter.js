import React, { Component } from 'react'

export class Filter extends Component {
  render() {
    return (
      <div style={{ border: "1px solid tomato" }}>
        <h1>Filter</h1>
      </div>
    )
  }
}

export default Filter
