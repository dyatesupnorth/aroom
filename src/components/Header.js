import React, { Component } from "react";

export class Header extends Component {
  render() {
    return (
      <div style={{ border: "1px solid tomato" }}>
        <h1>Header</h1>
      </div>
    );
  }
}

export default Header;
