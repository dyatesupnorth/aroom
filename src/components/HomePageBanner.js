import React, { Component } from "react";

export class HomePageBanner extends Component {
  render() {
    return (
      <div style={{ border: "1px solid tomato" }}>
        <h1>HomePageBanner</h1>
      </div>
    );
  }
}

export default HomePageBanner;
