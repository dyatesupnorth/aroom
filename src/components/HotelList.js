import React, { Component } from "react";

export class HotelList extends Component {
  render() {
    return (
      <div style={{ border: "1px solid tomato" }}>
        <h1>HotelList</h1>
      </div>
    );
  }
}

export default HotelList;
