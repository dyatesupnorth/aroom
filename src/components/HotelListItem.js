import React from "react";
import ListGroup from "react-bootstrap/ListGroup";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faStar } from "@fortawesome/free-solid-svg-icons";
export const renderStars = noOfStars => {
	console.log("​noOfStars", noOfStars)
  let stars = [];
	
  for (let i = 0; i < noOfStars; i++) {
		console.log("​i", i)
    stars.push(<FontAwesomeIcon icon={faStar} />);
  }
  console.log("​stars", stars)
  return stars;
};
export const HotelListItem = props => {
  return (
    <ListGroup.Item className="list-group-item-action flex-column align-items-start">
      <div className="d-flex w-100 justify-content-between">
        <h5 className="mb-1">{props.name}</h5>
      </div>
      <p className="mb-1">{renderStars(props.starRating)}</p>
      <small>
        {!props.facilities || props.facilities.length === 0 ? (
          ''
        ) : (
          <span>
            <ul className="list-inline">
              {props.facilities.map(facility => {
                return (
                  <li className="list-inline-item" key={facility}>
                    {facility}
                  </li>
                );
              })}
            </ul>
          </span>
        )}
      </small>
    </ListGroup.Item>
  );
};

export default HotelListItem;
