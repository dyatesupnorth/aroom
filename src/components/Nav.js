import React, { Component } from "react";

export class Nav extends Component {
  render() {
    return (
      <div style={{ border: "1px solid tomato" }}>
        <h1>Nav</h1>
      </div>
    );
  }
}

export default Nav;
