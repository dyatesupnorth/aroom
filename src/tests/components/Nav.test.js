import React from "react";
import { shallow } from "enzyme";
import { Nav } from "../../components/Nav";

test("should render HotelListItem", () => {
  const wrapper = shallow(<Nav />);
  expect(wrapper).toMatchSnapshot();
});
