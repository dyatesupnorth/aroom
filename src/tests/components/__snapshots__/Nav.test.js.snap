// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HotelListItem 1`] = `
<div
  style={
    Object {
      "border": "1px solid tomato",
    }
  }
>
  <h1>
    Nav
  </h1>
</div>
`;
